lib/faraday/utils.rb

```diff
@@ -95,7 +95,14 @@ def to_hash() ::Hash.new.update(self) end
 
       def parse(header_string)
         return unless header_string && !header_string.empty?
-        header_string.split(/\r\n/).
+
+        headers = header_string.split(/\r\n/)
+
+        # Find the last set of response headers.
+        start_index = headers.rindex { |x| x.match(/^HTTP\//) }
```

Comment:
I'm getting an issue where this returns `nil` when replaying a cassette from WebMock, because there's no `HTTP ...` line in the replayed headers. Is this a bug in Faraday or in WebMock?